scripts/run_e2e_tests.py

```diff
@@ -0,0 +1,472 @@
+    """Change constant file based on the running mode. Only the `DEV_MODE` line
+    should be changed.
+
+    Args:
+        constant_file: str. File path to the constant file.
+        dev_mode: bool. Represents whether the program is running on dev
+            mode.
+    """
+    pattern = '"DEV_MODE": .*'
+    replace = '"DEV_MODE": %s' % (
+        'true' if dev_mode else 'false')
+    common.inplace_replace_file(constant_file, pattern, replace)
+
+
+def tweak_feconf_file(feconf_file_path, enable_community_dashboard):
+    """Change feconf.py file based on whether the community dashboard is
+    enabled.
+
+    Args:
+        feconf_file_path: str. Path to the feconf.py file.
+        enable_community_dashboard: bool. Represents whether community
+            dashboard is enabled.
+    """
+    pattern = 'COMMUNITY_DASHBOARD_ENABLED = .*'
+    replace = 'COMMUNITY_DASHBOARD_ENABLED = %s' % enable_community_dashboard
+    common.inplace_replace_file(feconf_file_path, pattern, replace)
+
+
+def run_webdriver_manager(parameters):
+    """Run commands of webdriver manager.
+
+    Args:
+        parameters: list(str). A list of parameters to pass to webdriver
+            manager.
+    """
+    web_driver_command = [common.NODE_BIN_PATH, WEBDRIVER_MANAGER_BIN_PATH]
+    web_driver_command.extend(parameters)
+    python_utils.PRINT(common.run_cmd(web_driver_command))
+
+
+def setup_and_install_dependencies(skip_install):
+    """Run the setup and installation scripts."""
+    if not skip_install:
+        install_third_party_libs.main()
+    setup.main(args=[])
+    setup_gae.main(args=[])
+    if os.getenv('TRAVIS'):
+        install_chrome_on_travis.main(args=[])
+
+
+def build_js_files(dev_mode):
+    """Build the javascript files.
+
+    Args:
+        dev_mode: bool. Represents whether run the related commands in dev
+            mode.
+    """
+    tweak_constant_file(CONSTANT_FILE_PATH, dev_mode)
+    if not dev_mode:
+        python_utils.PRINT('  Generating files for production mode...')
+    else:
+        # The 'hashes.json' file is used by the `url-interpolation` service.
+        if not os.path.isfile(HASHES_FILE_PATH):
+            with python_utils.open_file(HASHES_FILE_PATH, 'w') as hash_file:
+                hash_file.write('{}')
+        try:
+            common.run_cmd(
+                [common.NODE_BIN_PATH, WEBPACK_BIN_PATH, '--config',
+                 'webpack.dev.config.ts'])
+        except subprocess.CalledProcessError as error:
+            python_utils.PRINT(error.output)
+            sys.exit(error.returncode)
+    build.main(args=(['--prod_env'] if not dev_mode else []))
+
+
+def tweak_webdriver_manager():
+    """webdriver-manager (version 13.0.0) uses `os.arch()` to determine the
+    architecture of the operation system, however, this function can only be
+    used to determine the architecture of the machine that compiled `node`
+    (great job!). In the case of Windows, we are using the portable version,
```

Comment:
I smiled a bit while reading this, but I think we might want to take some of the more personal bits out. Suggest eliding "(great job!)", and also changing "While clearly ..." to "Unfortunately, webdriver-manager seems to assume that Windows wouldn't run on the ia32 architecture, so its help function returns null for this, which means that the application has no idea about where to download the correct version."

What do you mean by "its own help function will return null for this", btw? Is os.arch() the "help function" you're referring to or is it something else?

Also, thank you for taking the time to explain clearly what's going on here and provide the links -- it helps a lot!

Reply:
Updated

The help function is the one decides the download link based on the architecture.